### PR TITLE
read only indicator

### PR DESCRIPTION
**Pull request by sbromberger**
Rebased #2862.

### CODE REVIEW

**Comment by sbromberger** [comment]
Do we want read-only indicators in bufferline as well? It's an easy add.

File: helix-view/src/document.rs [hunk 1]
```diff
@@ -388,6 +399,15 @@ impl Document {
...
         doc.detect_indent_and_line_ending();
+        // this seems overly complex but should make it easier
+        // to extend doc.file_stat with more information in the future.
+        doc.write_permission = fs::metadata(path).map_or(WritePermission::Unknown, |f| {
+            if f.permissions().readonly() {
+                WritePermission::ReadOnly
+            } else {
+                WritePermission::Writable
+            }
+        });
```
Comment (archseer) [inline helix-view/src/document.rs:410]: Should probably be part of `doc.set_path` since a path can also be changed after opening?

Comment (sbromberger) [reply helix-view/src/document.rs:410]: What are the implications of changing a path after opening?

**Comment by yashpalgoyal1304** [comment]
Why did the author close this without any info?

**Comment by tjkirch** [comment]
Does anyone know if there's work ongoing for a read-only indicator, or a read-only flag/option for buffers?  I can't find any info about why this PR was closed or anything happening after it.

**Pull request closed without merging**